drivers/spi/spi_gd32.c

```diff
@@ -238,6 +252,59 @@ static int spi_gd32_transceive(const struct device *dev,
 	return ret;
 }
 
+static int spi_gd32_transceive(const struct device *dev,
+			       const struct spi_config *config,
+			       const struct spi_buf_set *tx_bufs,
+			       const struct spi_buf_set *rx_bufs)
+{
+	return spi_gd32_transceive_impl(dev, config, tx_bufs, rx_bufs, NULL);
+}
+
+#ifdef CONFIG_SPI_ASYNC
+static int spi_gd32_transceive_async(const struct device *dev,
+				     const struct spi_config *config,
+				     const struct spi_buf_set *tx_bufs,
+				     const struct spi_buf_set *rx_bufs,
+				     struct k_poll_signal *async)
+{
+	return spi_gd32_transceive_impl(dev, config, tx_bufs, rx_bufs, async);
+}
+#endif
+
+#ifdef CONFIG_SPI_GD32_INTERRUPT
+
+static void spi_gd32_complete(const struct device *dev, int status)
+{
+	struct spi_gd32_data *data = dev->data;
+	const struct spi_gd32_config *cfg = dev->config;
+
+	SPI_CTL0(cfg->reg) &= ~SPI_CTL0_SPIEN;
```

Comment:
Please consider to remove this line, gd32f450i_eval can not pass spi_flash with this.

Reply:
Could you tell me the details of the problem.

It disables spi after spi_context_wait_for_completion is finished, so it should be okay to remove it.

Reply:
It seems work correctly even if this line removed.
Once, remove this line.

Reply:
Use below debug patch to trace gd32 spi isr:
```
diff --git a/drivers/spi/spi_gd32.c b/drivers/spi/spi_gd32.c
index ca033a9bb4..6d2a8628b5 100644
--- a/drivers/spi/spi_gd32.c
+++ b/drivers/spi/spi_gd32.c
@@ -286,7 +286,9 @@ static void spi_gd32_complete(const struct device *dev, int status)
 {
        struct spi_gd32_data *data = dev->data;
        const struct spi_gd32_config *cfg = dev->config;
+       printk("complete\n");
 
+       SPI_CTL0(cfg->reg) &= ~SPI_CTL0_SPIEN;
        SPI_CTL1(cfg->reg) &= ~(SPI_CTL1_RBNEIE | SPI_CTL1_TBEIE | SPI_CTL1_ERRIE);
 
        spi_context_complete(&data->ctx, status);
@@ -298,15 +300,19 @@ static void spi_gd32_isr(struct device *dev)
        struct spi_gd32_data *data = dev->data;
        int err = 0;
 
+       printk("isr_1: %x\n", SPI_STAT(cfg->reg));
        if ((SPI_STAT(cfg->reg) & SPI_GD32_ERR_MASK) != 0) {
                err = spi_gd32_get_err(cfg);
        } else {
+               printk("exchange\n");
                err = spi_gd32_frame_exchange(dev);
        }
+       printk("isr_2\n");
 
        if (err || !spi_gd32_transfer_ongoing(data)) {
                spi_gd32_complete(dev, err);
        }
+       printk("isr_3\n");
 
        SPI_STAT(cfg->reg) = 0;
 }

```

On gd32f450i_eval, we can get below output with spi_flash example. 

There have an extral `isr` after the `complete` funcation have been called. 
In the extral `isr`, it halt at `exchange` function.
Why `isr` halt at `exchange` function? Because we disabled i2c at `complete` function, `RBNE` will never be set again.

```
isr_1: 2
exchange
isr_2
isr_3
isr_1: 2
exchange
isr_2
isr_3
isr_1: 2
exchange
isr_2
isr_3
isr_1: 2
exchange
isr_2
complete
isr_3
isr_1: 2
exchange

```

After remove the spi disable line,  there have two `complete` funcation call.
```
isr_1: 2
exchange
isr_2
isr_3
isr_1: 2
exchange
isr_2
isr_3
isr_1: 2
exchange
isr_2
isr_3
isr_1: 2
exchange
isr_2
complete
isr_3
isr_1: 2
exchange
isr_2
complete
isr_3
```

We can see there have an extral `TBE` interrupt event though we have cleared the `SPI_CTL1_TBEIE` bit.

To cover this situation, maybe we can add a on_going check before enter `exchange` function and prevent re-enter `complete` function.

Reply:
Thank you for the explanation.
I got that interrupts are not received correctly by stopping the interrupt SPI in isr.
Eventually, Stop the SPI immediately before completing the transfer with spi_context_release (). 
So, removing that line from spi_gd32_complete () seems to be well.